tests/test_rectangle.py

```diff
@@ -118,7 +118,9 @@ def test_rectangular_section_plastic():
 def test_rectangular_section_warping():
     """Test rectangular section warping properties."""
     check.almost_equal(rectangle_section.section_props.j, 2861002, rel=2 * warp_tol)
-    check.almost_equal(rectangle_section.section_props.gamma, 3.177234e08, rel=tol)
+    check.almost_equal(
+        rectangle_section.section_props.gamma, 3.177234e08, rel=5 * warp_tol
```

Comment:
This difference is puzzling, all other warping results appear to be exactly the same with your changes.

I've narrowed this down to changing `n=6` to `n=4` Gauss points in `shear_warping_integrals()`. If I'm being honest, it's a very long time since I've thought about quadrature requirements given the functions we're integrating (5+ years?) I'm wondering what your thoughts are on this one? 

Do you think `n=4` is sufficient given that the results for `gamma` have changed? Given my *current* lack of insight on this I'm hesitant on making any changes that break the tests results and am leaning towards keeping `n=6` here.

Reply:
Theoretically, the shape function used in T6 is complete up to second order, which should be accurately integrated using a 3-point quadrature. In practical FEA, considering material nonlinearity, often a 4-point quadrature is chosen for T6, which is enough, there is no need to use higher order one because when material nonlinearity is involved, both quadrature won't be accurate/exact. Mesh refinement is require anyway for better results.

In this particular case, integrating N*N^T requires 2nd order quadrature, integrating B*B^T requires 1st order quadrature. And we only have elastic cases here, so 4-point quandrature is very sufficient.

I am not surprised by the changes. Even with the original implementation, mesh refinement would give an omega that is closer to what we have now, instead of the reference value.

For example, changing mesh size to 2 alone,

```py
rectangle_geometry.create_mesh(mesh_sizes=2)
```
leads to failure

```text
FAILURE: check 317541841.2903414 == pytest.approx(317723400.0, rel=1e-06, abs=None)
```

How the reference value was obtained? Is it numerial result or theoretical value?

Reply:
Thanks for the refresher @TLCFEM, I am happy to keep the changes you have suggested. From memory the rectangular and angle results were obtained from Strand7, so they are numerical results. There was also no way to control the mesh size at the time so they are by no means refined. I no longer have access to Strand7 to confirm, but I am happy with the results as is :)

Reply:
I wanted to use 3-point quadrature but it failed so many tests. I do not have time to refactor those for the moment so I do not touch it. Those reference values may need to be updated by using a denser mesh, if they are numerical results.

For `fea.py`, there is nothing much can be done that may significantly optimise the performance for the moment. I'll check other files.